### PR TITLE
feat: remove line element from scatter controller default config

### PR DESCRIPTION
**Pull request by thabarbados**

### CODE REVIEW

File: src/core/core.controller.js [hunk 1]
```diff
@@ -420,7 +420,12 @@ class Chart {
         meta.controller.linkScales();
       } else {
         const ControllerClass = registry.getController(type);
-        const {datasetElementType, dataElementType} = defaults.datasets[type];
+        let {datasetElementType, dataElementType} = defaults.datasets[type];
+
+        if (type === 'scatter' && dataset.showLine) {
```
Comment (dangreen) [inline src/core/core.controller.js]: Base class shouldn't know about child types.

**Review by dangreen — CHANGES_REQUESTED** [review]

File: src/core/core.controller.js (hunk 1 above)
Comment (thabarbados) [reply src/core/core.controller.js]: fixed

File: src/controllers/controller.scatter.js [hunk 2]
```diff
@@ -1,7 +1,133 @@
...
+        datasetElementType: 'line',
+        dataElementType: 'point',
+        showLine: true,
+        fill: false,
+      };
+      this.chart.buildOrUpdateControllers(config);
+    }
+  }
+
+  update(mode) {
```
Comment (dangreen) [inline src/controllers/controller.scatter.js:24]: @thabarbados  Is this code copypasted from LineController?

Comment (dangreen) [reply src/controllers/controller.scatter.js:24]: If there is common code with LineController, maybe then move some to a common file with utils?  To dedupe code.

Comment (thabarbados) [reply src/controllers/controller.scatter.js:24]: @dangreen Yes, the source of this method is the LineController, but this is not a copy-paste. 
This code was copied from the LineController and modified for ScatterController.

**Review by dangreen — APPROVED** [review]

**Pull request closed without merging**